compiler/src/test/kotlin/io/spine/protodata/CompilerEventsTest.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata
+
+import com.google.common.truth.extensions.proto.ProtoTruth.assertThat
+import com.google.protobuf.BoolValue
+import com.google.protobuf.DescriptorProtos.FileOptions.JAVA_MULTIPLE_FILES_FIELD_NUMBER
+import com.google.protobuf.StringValue
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorRequest
+import io.spine.base.EventMessage
+import io.spine.option.OptionsProto.REQUIRED_FIELD_NUMBER
+import io.spine.option.OptionsProto.TYPE_URL_PREFIX_FIELD_NUMBER
+import io.spine.protodata.test.DoctorProto
+import io.spine.testing.Correspondences.type
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class `'CompilerEvents' should` {
```

Comment:
`internal`?

Reply:
I suggest skipping visibility modifiers for tests, just like we do in Java. I don't suppose anyone would ever try to use a test case class.

Reply:
If memory serves, when IDEA creates a test suite in Kotlin it does apply `internal`. In Java, when we skip `public` the class becomes `package-private` and is not visible from outside (e.g. when auto-importing in IDEA).

If we skip this it in Kotlin, and assuming a big number of test suites, it is going to impact performance of the IDE sooner or later.

There is also a discipline aspect of having `internal` applied to tests. The code of tests _is_ equally important as production code. Therefore, it should be treated as such. We should not be sloppy about it.